src/qibolab/compilers/default.py

```diff
@@ -27,11 +23,15 @@ def rz_rule(gate: Gate, qubit: Qubit) -> PulseSequence:
     return PulseSequence([(qubit.drive.name, VirtualZ(phase=gate.parameters[0]))])
 
 
-def gpi2_rule(gate: Gate, qubit: Qubit) -> PulseSequence:
+def identity_rule(gate: Gate, qubit: SingleQubitNatives) -> PulseSequence:
+    """Identity gate skipped."""
+    return PulseSequence()
+
+
+def gpi2_rule(gate: Gate, qubit: SingleQubitNatives) -> PulseSequence:
     """Rule for GPI2."""
-    return qubit.native_gates.RX.create_sequence(
-        theta=np.pi / 2, phi=gate.parameters[0]
-    )
+    assert qubit.RX is not None
```

Comment:
We could raise a more readable error? It can be a single function called by all rules.

Reply:
Absolutely. I wanted to narrow the type for the type checker. I did not bother at all concerning the interface, and went for the simplest.

Reply:
Done in c52467f4a489349fe943b0808bd9c0931e1ae5c1